pkg/component/crddeployer/crd_test.go

```diff
@@ -0,0 +1,175 @@
+		testClient client.Client
+
+		unreadyCRD = &apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "myresources.mygroup.example.com",
+			},
+		}
+		readyCRD = &apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "myresources.mygroup.example.com",
+			},
+			Status: apiextensionsv1.CustomResourceDefinitionStatus{
+				Conditions: []apiextensionsv1.CustomResourceDefinitionCondition{
+					{Type: apiextensionsv1.Established, Status: apiextensionsv1.ConditionTrue},
+					{Type: apiextensionsv1.NamesAccepted, Status: apiextensionsv1.ConditionTrue},
+				},
+			},
+		}
+
+		validManifest   string
+		invalidManifest string
+
+		crdDeployer component.DeployWaiter
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		validManifest = `apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+    name: myresources.mygroup.example.com`
+		invalidManifest = `thisIsNotAValidManifest`
+
+		testClient = fakeclient.NewClientBuilder().WithScheme(kubernetesscheme.Scheme).Build()
+		mapper := meta.NewDefaultRESTMapper([]schema.GroupVersion{apiextensionsv1.SchemeGroupVersion})
+		mapper.Add(apiextensionsv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"), meta.RESTScopeRoot)
+		applier = kubernetes.NewApplier(testClient, mapper)
+		var err error
+		crdDeployer, err = NewCRDDeployer(testClient, applier, []string{validManifest})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(crdDeployer).ToNot(BeNil())
+	})
+
+	Describe("#Deploy", func() {
+		It("should deploy a CRD", func() {
+			actualCRD := &apiextensionsv1.CustomResourceDefinition{}
+
+			Expect(crdDeployer.Deploy(ctx)).To(Succeed())
+
+			Expect(testClient.Get(ctx, client.ObjectKey{Name: readyCRD.Name}, actualCRD)).To(Succeed())
+			Expect(actualCRD.Name).To(Equal(readyCRD.Name))
+		})
+	})
+
+	Describe("#Destroy", func() {
+		It("should destroy a CRD", func() {
+			actualCRD := &apiextensionsv1.CustomResourceDefinition{}
+
+			Expect(testClient.Create(ctx, readyCRD)).To(Succeed())
+
+			Expect(crdDeployer.Destroy(ctx)).To(Succeed())
+
+			Expect(testClient.Get(ctx, client.ObjectKey{Name: readyCRD.Name}, actualCRD)).To(matchers.BeNotFoundError())
+		})
+	})
+
+	Describe("#MakeCRDNameMap", func() {
+		It("should return a map representing the CRD name map", func() {
+			crdNameToManifest, err := MakeCRDNameMap([]string{validManifest})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(crdNameToManifest).To(HaveKeyWithValue("myresources.mygroup.example.com", validManifest))
+		})
+
+		It("should throw an error when a non valid CRD is provided", func() {
+			crdMap, err := MakeCRDNameMap([]string{invalidManifest})
+			Expect(crdMap).To(BeNil())
+			Expect(err).To(MatchError(ContainSubstring("error unmarshaling JSON")))
+		})
+	})
+
+	Describe("#WaitUntilCRDManifestsReady", func() {
+		It("should return true because the CRD is ready", func() {
+			testClient := fakeclient.NewClientBuilder().
```

Comment:
Why do you initiate a new test client here?

Reply:
Sorry for the oversight :/
This is a remainder before I initialized with `BeforeEach`